stock_by_warehouse/models/product_template.py

```diff
@@ -0,0 +1,73 @@
+import json
+from odoo import api, fields, models, _
+
+UNIT = 'Product Unit of Measure'
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    qty_available_not_res = fields.Float(
+        string='Quantity On Hand Unreserved',
+        digits=UNIT,
+        compute='_compute_product_available_not_res',
+    )
+
+    warehouses_stock = fields.Text(
+        store=False, readonly=True,
+    )
+    warehouses_stock_recompute = fields.Boolean(store=False, readonly=False)
+
+    @api.onchange('warehouses_stock_recompute')
+    def _warehouses_stock_recompute_onchange(self):
+        if not self.warehouses_stock_recompute:
+            self.warehouses_stock_recompute = True
+            return
+        self.warehouses_stock = self._compute_get_quantity_warehouses_json()
+        self.warehouses_stock_recompute = True
+
+    @api.depends('product_variant_ids.qty_available_not_res')
+    @api.depends_context('warehouse', 'force_company')
```

Comment:
Why this?

Reply:
Because, now the computed field for the product quantities in [product.template](https://github.com/odoo/odoo/blob/13.0/addons/stock/models/product.py#L617) and [product.product](https://github.com/odoo/odoo/blob/13.0/addons/stock/models/product.py#L96), uses `@api.depends_context`, when using this decoration, it only recognizes the keys mentioned in it, for the calls that use context.

Also, it is necessary to filter by warehouses and companies, otherwise, it takes the general quantity for all warehouses

Reply:
@CarmenMiranda could you check @deivislaya's comment regarding "it is necessary to filter by warehouses and companies, otherwise, it takes the general quantity for all warehouses", please?

Reply:
@CarmenMiranda could you please review

Reply:
I'm not fully aware of the `@api.depends_contex` but as I see it on the @deivislaya examples is need it, but, also I saw that there was a [commit](https://github.com/odoo/odoo/commit/de2725c173cfe028c3c0f411e613ba732be14772) that removed the `warehouse` key from the `@api.depends_contex` 
 of the method `_compute_quantities`. So I'm not sure if that key really needs to be here or if it can be a problem with it.

Reply:
Yes, they eliminated it because, from the current code, they no longer need to obtain the search for warehouses in a single attempt, in the field